File: src/main/java/top/criwits/scaleda/verilog/psi/references/ModuleInstantiationReference.kt
//package top.criwits.scaleda.verilog.psi.references
//
//import com.intellij.psi.PsiElementResolveResult
//import com.intellij.psi.PsiReferenceBase
//import com.intellij.psi.ResolveResult
//import top.criwits.scaleda.verilog.psi.factory.nodes.ModuleDeclarationPsiNode
//import top.criwits.scaleda.verilog.psi.factory.nodes.ModuleInstantiationPsiNode
//import top.criwits.scaleda.verilog.utils.*;
//
//class ModuleInstantiationReferenceOld(element: ModuleInstantiationPsiNode) :
//    PsiReferenceBase.Poly<ModuleInstantiationPsiNode>(
//        element,
//        element.getHoldPsiNodeRelativeTextRange(),
//        true
//    ) {
//
//    override fun multiResolve(incompleteCode: Boolean): Array<ResolveResult> {
//        return FileUtilsJava.getAllVerilogFilesList(myElement.project)
//            .flatMap { it.availableNamedElementsLists }
//            .filterIsInstance(ModuleDeclarationPsiNode::class.java)
//            .filter { it.name == myElement.getHoldPsiNode()?.name }
//            .map { PsiElementResolveResult(it) }
//            .toTypedArray()
//    }
//
//    override fun getVariants(): Array<Any> {
//        return emptyArray()
//    }
//
//}

File: src/main/scala/top/criwits/scaleda/idea/verilog/references/ModuleInstantiationReference.scala
package top.criwits.scaleda
package idea.verilog.references

import verilog.psi.factory.nodes.ModuleDeclarationPsiNode

import com.intellij.psi.{PsiElementResolveResult, PsiReferenceBase, ResolveResult}
import top.criwits.scaleda.idea.verilog.utils.FileUtils

class ModuleInstantiationReference(element: ModuleInstantiationPsiNode)
  extends PsiReferenceBase.Poly[ModuleInstantiationPsiNode](element, element.getHoldPsiNodeRelativeTextRange, true) {
  override def multiResolve(incompleteCode: Boolean): Array[ResolveResult] =
    FileUtils.getAllVerilogFiles(myElement.getProject)
      .flatMap(it => it.getAvailableNamedElementsScala)
      .filter(it => it.isInstanceOf[ModuleDeclarationPsiNode])
      .filter(x => x.getName == myElement.getHoldPsiNode.getName)
      .map(it => new PsiElementResolveResult(it))
      .toArray


  override def getVariants = Array()
}